src/maxp.cc

```diff
@@ -29,6 +29,23 @@ bool OpenTypeMAXP::Parse(const uint8_t *data, size_t length) {
     return Error("numGlyphs is 0");
   }
 
+  // Per https://learn.microsoft.com/en-gb/typography/opentype/spec/maxp,
+  // the only two 'maxp' version numbers are 0.5 (for CFF/CFF2) and 1.0
+  // (for TrueType).
+  if (version == 0x00005000) {
+    this->version_1 = false;
+    return true;
+  }
+
+  if (version != 0x00010000) {
+    Warning("Unrecognized version %08x, attempting to fix", version);
+    // If there's at least enough data to read as a version 1.0 table,
+    // we'll try that.
+    if (length >= 32) {
+      version = 0x00010000;
+    }
```

Comment:
I think this check is unnecessary. The table reads all do bounding checks. I’d simplify the code from here below to try to read the data and sets `version_1` based on whether this was successful or not.